### ISSUE TITLE
Inputlocation for Spring Applications 

### ISSUE THREAD

**Issue by Liyw979** [soot-oss/SootUp#759]
version: 1.1.2

usage: PathBasedAnalysisInputLocation("D:\code_sample\sidemo-jar")

![image](https://github.com/soot-oss/SootUp/assets/16919236/7961e2f0-1b6c-4f79-9dfb-088c622d91ef)

![image](https://github.com/soot-oss/SootUp/assets/16919236/c430a611-081d-4505-a2df-e624517432e1)

**Comment by JonasKlauke** [comment]
Yeah that happens with Spring Application jars. Because Spring uses its own class loader. The application classes will be in the Boot-Inf/classes folder. Can you test it on the current development state. I think we have changed the way of the Fully Qualified Name generation.

**Comment by Liyw979** [comment]
It would be much better to have nightly builds, any plan for this? @JonasKlauke

**Comment by JonasKlauke** [comment]
yeah an issue was opened to publish snapshot releases on maven everytime we merge a PR into dev. #760

**Comment by Liyw979** [comment]
Just pull the current dev branch. still seeing this
![image](https://github.com/soot-oss/SootUp/assets/16919236/2a078913-3b9e-49cf-a903-e7ef0f546db7)

**Comment by Liyw979** [comment]
Looks like we are now getting class signatures from the file path.

`D:\code_sample\sidemo-jar` is a Directory, which triggles `DirectoryBasedAnalysisInputLocation` with `walkDirectory`

https://github.com/soot-oss/SootUp/blob/5d63bd86f0c1751a6706cb57bccac2c91073fd51/sootup.java.bytecode/src/main/java/sootup/java/bytecode/inputlocation/PathBasedAnalysisInputLocation.java#L173-L175

**Comment by Liyw979** [comment]
https://github.com/soot-oss/SootUp/blob/5d63bd86f0c1751a6706cb57bccac2c91073fd51/sootup.java.bytecode/src/main/java/sootup/java/bytecode/frontend/AsmClassSource.java#L126-L130

We can use `classNode.name` to obtain the correct classType.

![image](https://github.com/soot-oss/SootUp/assets/16919236/330a56aa-356d-45b8-92f4-829350542b61)

I will make a PR for this.

**Comment by JonasKlauke** [comment]
We discussed this yesterday. 
At the moment SootUp uses the path as FQN. The reason for it is that if you request a class you pass a classtype. The inputlocation has to look up the class and transform it to Jimple, since we are doing lazy loading of classes. But if the FQN does not match the path the input location can not find the class. At the state in which you walk through the folder, you don't know the correct FQN without doing some expensive ASM requests. So I had the idea that we have to save preambles to jars inside a jars or to the Boot-INF/classes folder, so the input location can check multiple locations to find the FQN.

If you know a nice way how we get the FQN while going through the folders, share it with us

**Comment by Liyw979** [comment]
https://github.com/soot-oss/SootUp/blob/5d63bd86f0c1751a6706cb57bccac2c91073fd51/sootup.java.bytecode/src/main/java/sootup/java/bytecode/inputlocation/PathBasedAnalysisInputLocation.java#L182-L199
You mean this ? 
We can walk through the dir recursively and cache the paths in advance like `xxx/org/sec/sidemo..../xxx.class`
And when some `signature` passed in like `org.sec.sidemo....Controller` , we can search the cache to find what path contains `org/sec/sidemo..../Controller.class`.

**Comment by swissiety** [comment]
Feel free to extend SootUp and build a custom Analysisinputlocation which creates a classtype -> classsource / path index like you described with the real fq classname that was found inside the .class file -The current implementation of SootUp models the behaviour of the default classloader i.e. uses the classpath and accesses the class files according to its convention. Maybe someone is interested in creating a more precise/performant AnalysisInputLocation designated for the SpringBoot Classloader as well.

**Comment by Liyw979** [comment]
I get this. Thank you two!
